src/einspline/tests/test_one.cpp

```diff
@@ -25,15 +26,15 @@
 
 using std::string;
 
-
 TEST_CASE("double_1d_natural", "[einspline]")
 {
+  // two point case
   Ugrid x_grid;
   x_grid.start = 1.0;
   x_grid.end = 10.0;
   x_grid.num = 2;
 
-  double data[2];
+  double data[3];
```

Comment:
The test was originally written with just 2.  This just allows me to run the new tests.

Reply:
I assume test code shows expected behavior and proper usage.  I elected not to go with the fixes I wrote but simply assert on bad inputs and remove them from the tests since they result in overflow reads that trip ASAN.  Since the QMCPACK lore is that never should a value outside of the BC be requested and that is clearly the way einspline as a whole was implemented. If we want to replace it with well designed C++ library that is something we can look for or do.

Reply:
I'm thinking of just removing the 2 point test.

Reply:
If you believe its incorrect we should.

Reply:
I agree with you that the caller code should handle the protection.
For this test particularly, do you remember where data[2] was accessed in the code? Is it the place where splines coefs are solved?

Reply:
I think the find_coefs_xxx routines

Reply:
and from there the solve

Reply:
And then in the solve.  By inspection it seems like the solver might actually work for two data points.

Reply:
So after you changed 10 to 9.9999999, does the test pass ASAN?

Reply:
yes